frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionPicker.jsx

```diff
@@ -51,7 +49,11 @@ function QuestionPicker({
 
   const handleSearchTextChange = e => setSearchText(e.target.value);
 
-  const collections = (collection && collection.children) || [];
+  const allCollections = (collection && collection.children) || [];
+  const isDashboardInPublicCollection = !dashboard.collection.is_personal;
+  const collections = isDashboardInPublicCollection
+    ? allCollections.filter(collection => !collection.is_personal)
```

Comment:
maybe this could be easier to understand 🤔 . Any thoughts?
```suggestion
    ? allCollections.filter(collection => isInPublicCollection(collection))
```

Reply:
yeah, this is a good change. It's more explicit and easier to understand for someone who hasn't been to this part of the code before

Reply:
Fixed in df8f53ce705c0e632a1c64373d6be0cc3dfc0cbb